src/edu/washington/escience/myria/coordinator/catalog/MasterCatalog.java

```diff
@@ -1561,6 +1574,48 @@ protected Object job(final SQLiteConnection sqliteConnection) throws CatalogExce
   }
 
   /**
+   * Update the partition function of an ingested dataset.
+   * 
+   * @param key the relation key.
+   * @param howPartitioned how the dataset was partitioned.
+   * @throws CatalogException if there is an error in the catalog.
+   */
+  public void updateHowPartitioned(final RelationKey key, final HowPartitioned howPartitioned) throws CatalogException {
+    if (isClosed) {
+      throw new CatalogException("Catalog is closed.");
+    }
+
+    try {
+      queue.execute(new SQLiteJob<DatasetStatus>() {
+        @Override
+        protected DatasetStatus job(final SQLiteConnection sqliteConnection) throws CatalogException, SQLiteException {
+          try {
+            SQLiteStatement statement =
+                sqliteConnection
+                    .prepare("UPDATE stored_relations set how_partitioned=? WHERE user_name=? AND program_name=? AND relation_name=?");
+            try {
+              statement.bind(1, MyriaJsonMapperProvider.getMapper().writeValueAsString(howPartitioned));
+            } catch (JsonProcessingException e) {
+              throw new CatalogException(e);
+            }
+            statement.bind(2, key.getUserName());
+            statement.bind(3, key.getProgramName());
+            statement.bind(4, key.getRelationName());
+            statement.stepThrough();
+            statement.dispose();
+            statement = null;
+          } catch (final SQLiteException e) {
+            throw new CatalogException(e);
+          }
+          return null;
+        }
+      }).get();
+    } catch (InterruptedException | ExecutionException e) {
```

Comment:
You should reset interrupted status when you catch `InterruptedException` since you aren't rethrowing the `InterruptedException` (e.g., http://stackoverflow.com/questions/3976344/handling-interruptedexception-in-java):

`Thread.currentThread().interrupt()`